src/app/state/transaction/index.ts

```diff
@@ -60,6 +62,30 @@ export const transactionSlice = createSlice({
       state.error = action.payload
       state.active = false
     },
+    addPendingTransaction(state, action: PayloadAction<PendingTransactionPayload>) {
+      // TODO: Add NetworkType differentiation
+      const {
+        payload: { from, transaction },
+      } = action
+
+      if (!state.pendingTransactions[from]) {
+        state.pendingTransactions[from] = [transaction]
+      } else {
+        state.pendingTransactions[from] = [transaction, ...state.pendingTransactions[from]]
+      }
+    },
+    removePendingTransaction(state, action: PayloadAction<PendingTransactionPayload>) {
```

Comment:
Instead of consensusWatchBlocks: just remove pending transactions when their hash appears in transactionsLoaded